packages/open-api-gateway/src/project/codegen/components/client-settings.ts

```diff
@@ -0,0 +1,201 @@
+      );
+    } else {
+      // spec didn't change and forceGenerate is off
+      // load previously generated client config
+      const clientSettingsPathAbs = path.join(
+        project.outdir,
+        this.clientSettingsPath
+      );
+      logger.trace(`Reading client settings from ${clientSettingsPathAbs}`);
+      const existingClientConfig = tryReadFileSync(clientSettingsPathAbs);
+      if (existingClientConfig != null) {
+        const parsedClientConfig = JSON.parse(existingClientConfig);
+
+        // previously generated client settings
+        const prevClientLanguages = parsedClientConfig.clientLanguages;
+        const prevDocFormats = parsedClientConfig.documentationFormats;
+
+        const currClientLanguages = options.clientLanguages;
+        const currDocFormats = options.documentationFormats;
+
+        // keep only those lang/doc formats that were not present previously
+        this.clientLanguageConfigs = this.getClientLanguageConfigs(
+          currClientLanguages.filter(
+            (lang) => !prevClientLanguages.includes(lang)
+          ),
+          options.defaultClientLanguage
+        );
+        this.documentationFormatConfigs = this.getDocumentationFormatConfigs(
+          currDocFormats.filter((format) => !prevDocFormats.includes(format))
+        );
+      } else {
+        logger.debug(
+          `No generated client settings file found at ${clientSettingsPathAbs}. All languages and docs will be generated`
+        );
+        this.clientLanguageConfigs = this.getClientLanguageConfigs(
+          options.clientLanguages,
+          options.defaultClientLanguage
+        );
+        this.documentationFormatConfigs = this.getDocumentationFormatConfigs(
+          options.documentationFormats
+        );
+      }
+    }
+  }
+
+  getClientLanguageConfigs(
+    clientLanguages: ClientLanguage[],
+    defaultLanguage: ClientLanguage
+  ): ClientLanguageConfig[] {
+    const values = Object.values(ClientLanguage);
+
+    return values.map((currLang) => {
+      return <ClientLanguageConfig>{
+        clientLanguage: currLang,
+        isDefault: defaultLanguage === currLang,
+        generate: clientLanguages.includes(currLang),
+      };
+    });
+  }
+
+  getDocumentationFormatConfigs(
+    documentationFormats: DocumentationFormat[]
+  ): DocumentationFormatConfig[] {
+    const values = Object.values(DocumentationFormat);
+
+    return values.map((currFormat) => {
+      return <DocumentationFormatConfig>{
+        documentationFormat: currFormat,
+        generate: documentationFormats.includes(currFormat),
+      };
+    });
+  }
+
+  synthesize(): void {
+    // store the client config
+    // this helps to optimize to re-generate projects when new language/doc format is
+    // introduced while the spec doesn't change
+    logger.trace(`Generating settings file to ${this.clientSettingsPath}`);
+    const clientSettingsJsonFile = new JsonFile(
+      this.project,
+      this.clientSettingsPath,
```

Comment:
Do we need to add a gitignore for this file?

Reply:
good call. added